### ISSUE TITLE
If main.py exists but MLProject does not, use main.py as default entry point [FR]

### ISSUE THREAD

**Issue by andyk** [mlflow/mlflow#2967]
## Willingness to contribute
The MLflow Community encourages new feature contributions. Would you or another member of your organization be willing to contribute an implementation of this feature (either as an MLflow Plugin or an enhancement to the MLflow code base)?

- [x] Yes. I can contribute this feature independently.
- [x ] Yes. I would be willing to contribute this feature with guidance from the MLflow community.

## Proposal Summary

Today, if I execute `mlflow run .` in a directory where an `MLProject` file does *not* exist but a `main.py` file does exist, MLflow does returns an error that it cannot find the default entry point "main" (since no entry points are defined anywhere).

Thus, I have to use the entry-point flag in this case, as in `mlflow run . --entry-point main.py`. 

I propose that if a file named `main.py` exists and one named `MLProject` does not, then MLflow run should run `python main.py` instead of returning an error.

## Motivation
- What is the use case for this feature?

This would make it easier to use the MLflow CLI to run projects that don't want to take the relatively heavy weight step of adding an MLProject file to their source repo.

- Why is this use case valuable to support for MLflow users in general?

This provides an extremely light weight path for folks to make their project/git repo a runnable entity via MLflow: just add main.py to your project, or if you already have one, you're already done.

- Why is this use case valuable to support for your project(s) or organization?

I'm writing a tool that treats directories as runnable things, and I'm using MLflow to do that but I don't want to force people who use my tool to learn about MLflow in order to start using it! So I don't want to force them to add an MLProject file to their project directory, I'd rather tell them they can simply create a main.py and then MLflow (and my tool) will run their project automatically. And then I can point to the parameter handling flags as the standard way to handle passing params.

This lowers the friction to having many more projects become runnable by MLflow, which might spread the word and adoption of MLflow!

- Why is it currently difficult to achieve this use case? (please be as specific as possible about why related MLflow features and components are insufficient)

This isn't much code, I think it would be a few lines of code [here](https://github.com/mlflow/mlflow/blob/4f576bbcb889275fda303fe531344ff6bb160c9d/mlflow/projects/_project_spec.py#L99) plus some tweaks to the docs and CLI doc strings.

### What component(s), interfaces, languages, and integrations does this feature affect?
Components 

- [x] `area/projects`: MLproject format, project running backends

Interfaces
- [x] `area/uiux`: Front-end, user experience, JavaScript, plotting

**Comment by smurching** [comment]
@andyk makes sense to me! Would be happy to review a PR adding support for this. Note that the current alternative would be to `mlflow run <project-url> -e main.py`, but that workaround is pretty hard to discover & there don't seem to be any negative long-term consequences of making it easy to e.g. `mlflow run https://github.com/yunjey/stargan`. Also agree that main.py is a reasonable default entry point if there's no MLproject file (as you mentioned in offline discussion, it matches the `main` default entry point when there is an MLproject file) - there's also at least one example of a popular project that exposes a main.py (StarGAN), and github search suggests that [other projects do too](https://github.com/search?q=%22main.py%22&type=Repositories).
